DllImportGenerator/designs/Pipeline.md

```diff
@@ -139,6 +139,21 @@ These various scenarios have different levels of support for these three feature
 | non-blittable array marshalling in a P/Invoke | unsupported | unsupported (supportable with https://github.com/dotnet/runtime/issues/25423) | unuspported |
 | non-blittable array marshalling not in a P/Invoke | unsupported | unsupported | unuspported |
 
+### `SetLastError=true`
+
+The stub code generation also handles [`SetLastError=true`][SetLastError] behaviour. This configuration indicates that system error code ([`errno`](https://en.wikipedia.org/wiki/Errno.h) on Unix, [`GetLastError`](https://docs.microsoft.com/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror) on Windows) should be stored after the native invocation, such that it can be retrieved using [`Marshal.GetLastWin32Error`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.marshal.getlastwin32error).
+
+This means that, rather than simply invoke the native method, the generated stub will:
+
+1. Clear the system error by setting it to 0
+2. Invoke the native method
+3. Get the system error
+4. Set the stored error for the P/Invoke (accessible via `Marshal.GetLastWin32Error`)
```

Comment:
So (4) is where we need the new API, yes?

Reply:
And (1).

Reply:
Also (3) for getting the system error.